src/nv_ingest/modules/sources/message_broker_task_source.py

```diff
@@ -124,22 +108,34 @@ def process_message(job: Dict, ts_fetched: datetime) -> ControlMessage:
         do_trace_tagging = tracing_options.get("trace", False)
         ts_send = tracing_options.get("ts_send", None)
         if ts_send is not None:
-            # ts_send is in nanoseconds
+            # ts_send is in nanoseconds.
             ts_send = datetime.fromtimestamp(ts_send / 1e9)
         trace_id = tracing_options.get("trace_id", None)
 
         response_channel = f"{job_id}"
 
-        df = cudf.DataFrame(job_payload)
-        message_meta = MessageMeta(df=df)
+        df = pd.DataFrame(job_payload)
+        control_message.payload(df)
 
-        control_message.payload(message_meta)
         annotate_cm(control_message, message="Created")
         control_message.set_metadata("response_channel", response_channel)
         control_message.set_metadata("job_id", job_id)
 
+        # For each task, build a ControlMessageTask instance and add it.
         for task in job_tasks:
-            control_message.add_task(task["type"], task["task_properties"])
+            task_id = task.get("id", str(uuid.uuid4()))
+            task_type = task.get("type", "unknown")
+            task_props = task.get("task_properties", {})
+            if not isinstance(task_props, dict):
+                task_props = task_props.model_dump()
+
+            task_obj = ControlMessageTask(
+                id=task_id,
+                type=task_type,
+                properties=task_props,
+            )
+            logger.info(task_obj.model_dump())
```

Comment:
Should we remove this or change the level to `logger.debug`?

Reply:
Removed